### PR TITLE
NetSim: Teacher view dropdown bug

### PR DESCRIPTION
**Pull request by islemaster**
@sfilman reported a bug in the new Teacher View sent-by dropdown during our bug bash today ([original report](https://docs.google.com/document/d/1ggbAAhc9qX5DyrtMF53Bs7_BOQi96My6kws5UWldDAU/edit#bookmark=id.131whwr54xef)).  The dropdown had an `undefined` entry instead of a list of message senders.

image

The fix is trivial (I was referring to the wrong property name).  I've gone ahead and added some simple tests, too.  The tests make use of these libraries, which it may be useful to know about:
- [Enzyme](https://github.com/airbnb/enzyme), the React test library for rendering components and querying the results.
- [chai-enzyme](https://github.com/producthunt/chai-enzyme) which adds BDD-style Chai assertions that use Enzyme queries and produces useful output when those assertions fail.

### CODE REVIEW

File: apps/src/netsim/NetSimLogBrowserFilters.jsx [hunk 1]
```diff
@@ -159,7 +159,7 @@ const SentByDropdown = React.createClass({
         <option value="none">
           {this.props.i18n.logBrowserHeader_sentByAnyone()}
         </option>
-        {_.uniq(this.props.logRows.map(row => row.sentBy))
+        {_.uniq(this.props.logRows.map(row => row['sent-by']))
```
Comment (islemaster) [inline apps/src/netsim/NetSimLogBrowserFilters.jsx:162]: This is the actual bugfix.

File: apps/test/util/configuredChai.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import chai from 'chai';
 import chaiSubset from 'chai-subset';
 import chaiEnzyme from 'chai-enzyme';
-chai.use(chaiEnzyme);
+chai.use(chaiEnzyme());
```
Comment (islemaster) [inline apps/test/util/configuredChai.js:5]: @pcardune added chai-enzyme in #9629 but didn't use it anywhere.  Turns out we were installing it incorrectly.  This is the correct approach, according to the [chai-enzyme docs](https://github.com/producthunt/chai-enzyme#setup).

**Comment by davidsbailey** [comment]
The fix itself looks good. I'll take a closer look at the tests and chai/enzyme today.

**Comment by islemaster** [comment]
I'm going to go ahead and merge then, since the test changes are unit test only and are passing on CircleCI.

**Pull request merged**